third_party/pcsc-lite/naclport/server_clients_management/src/client_request_processor.cc

```diff
@@ -491,26 +576,31 @@ GenericRequestResult PcscLiteClientRequestProcessor::SCardDisconnect(
                       DebugDumpSCardDisposition(disposition_action));
   tracer.LogEntrance();
 
-  LONG return_code = SCARD_S_SUCCESS;
-  if (!s_card_handles_registry_.ContainsHandle(s_card_handle))
-    return_code = SCARD_E_INVALID_HANDLE;
-
-  if (return_code == SCARD_S_SUCCESS) {
-    return_code = ::SCardDisconnect(s_card_handle, disposition_action);
-
-    // Catch when PC/SC-Lite core, unlike us, thinks the handle doesn't exist.
-    if (return_code == SCARD_E_INVALID_HANDLE)
-      OnSCardHandleRevoked(s_card_handle);
-  }
+  LONG return_code = DisconnectCard(&s_card_handle, disposition_action);
 
   tracer.AddReturnValue(DebugDumpSCardReturnCode(return_code));
   tracer.LogExit();
 
-  if (return_code == SCARD_S_SUCCESS)
-    s_card_handles_registry_.RemoveHandle(s_card_handle);
   return ReturnValues(return_code);
 }
 
+LONG PcscLiteClientRequestProcessor::DisconnectCard(LPSCARDHANDLE s_card_handle,
```

Comment:
nit: No need to pass the handle as a pointer here; let's use `SCARDHANDLE` instead.